src/r5py/util/memory_footprint.py

```diff
@@ -60,29 +60,82 @@ def share_of_ram(share=0.8, leave_at_least=(2 * (2**10))):
     return share_of_ram
 
 
-def max_memory(max_memory):
-    """Interpret the config parameter --max-memory."""
+def _parse_max_memory_string(max_memory):
+    """
+    Extract maximum memory value and unit from text input.
+
+    Arguments
+    ---------
+    max_memory : str
+        Input text from the config parameter --max-memory.
+
+    Returns
+    -------
+    tuple: a tuple containing
+        - value (float): Amount of memory to be allocated in a given unit.
+        - unit (str): The unit of memory.
+    """
     try:
-        matches = re.match(r"(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[%MGT])?", max_memory)
+        matches = re.match(r"(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[^0-9])?", max_memory)
```

Comment:
The regex should only match to the entire string. So, it should read `r"^(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[^0-9])?$"`

(Sorry, this is my bad - just tried it out the first time now that you submitted the PR)

Reply:
I’ll push the change myself, quickly, so you don’t have to wait another round.